### PR TITLE
Feature/issue 1108 - Implement in URL localization(/en) update on language switch (ua ↔ en)

### PR DESCRIPTION
**Pull request by weazy12**
## GitHub Tickets

* [Github Ticket](https://github.com/ita-social-projects/VictoryCenter-Back/issues/1108)

## How it looks

image

## CHECK LIST
- [ ]  New tests was added or existing was modified
- [ ]  Changes has severe impact on users
- [ ]  Changes has medium impact on users
- [ ]  Changes has light impact on users
- [ ]  Test covetage was updated
- [ ]  PR meets all conventions

## Summary by CodeRabbit

* **New Features**
  * Added a locale-aware Link component and a utility to generate locale-aware paths.
  * Locale hook now exposes the i18n instance in its public API.

* **Improvements**
  * Language detection now relies on the URL path only.
  * Language switching preserves path structure and query string, avoids no-op on same-language selection, and updates browser history.
  * Routing reorganized into explicit language-specific sections.

* **Tests**
  * Added coverage for language switching, path handling, query preservation, and link href behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by MykhailoKlapchuk — APPROVED** [review]
Looks good overall 👍
The locale sync logic is clean and well-tested. Please just double-check edge cases around default locale URLs and make sure external/absolute links aren’t affected by locale prefixing.

File: src/components/public/header/Header.test.tsx [hunk 1]
```diff
@@ -28,6 +28,19 @@ jest.mock('@/assets/icons/burger.svg', () => ({
...
+
+    return {
+        useTranslation: () => ({
+            t: (key: string) => headerUk[key] ?? key,
+        }),
+    };
+});
+
+jest.mock('@/hooks/common/use-locale/useLocale', () => ({
+    useLocale: () => ({ currentLanguage: 'uk' }), // або DEFAULT_LOCALE
```
Comment (snowflaki) [inline src/components/public/header/Header.test.tsx]: remove AI comment

File: src/components/public/header/Header.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { DropdownLink, DropdownMenu } from '@/components/public/dropdown-menu/Dr
 import { ReactComponent as BurgerIcon } from '@/assets/icons/burger.svg';
 import { LanguageSwitcher } from '@/components/public/language-switcher/LanguageSwitcher';
 import { useTranslation } from 'react-i18next';
-import { Link } from 'react-router-dom';
+import { Link } from '@/components/common/link/Link';
```
Comment (snowflaki) [inline src/components/public/header/Header.tsx]: use index file for Link component to make import cleaner

File: src/locales/i18n.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ i18n.use(initReactI18next).init({
     },
     detection: {
         // saving selected language
-        order: ['localStorage', 'navigator'],
+        order: ['path', 'localStorage', 'navigator'],
```
Comment (snowflaki) [inline src/locales/i18n.ts]: do we need to keep localStorage with new URL approach?

File: src/components/public/header/Header.test.tsx (hunk 1 above)
Comment (weazy12) [reply src/components/public/header/Header.test.tsx]: done

File: src/locales/i18n.ts (hunk 3 above)
Comment (weazy12) [reply src/locales/i18n.ts]: done, i think we also don`t need a naviagtor because we guided only by path so I remove navigator too.

File: src/components/public/header/Header.tsx (hunk 2 above)
Comment (weazy12) [reply src/components/public/header/Header.tsx]: done

File: src/components/common/link/Link.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { DEFAULT_LOCALE } from '@/const/common/locales';
+
+type AppLinkProps = Omit<LinkProps, 'to'> & {
+    to: string;
+};
+
+export const Link = ({ to, children, ...props }: AppLinkProps) => {
+    const { currentLanguage } = useLocale();
+
+    const getLocalizedPath = (path: string): typeof path => {
```
Comment (snowflaki) [inline src/components/common/link/Link.tsx]: move this function to utils

```tsx
import type { To } from 'react-router-dom';
import { DEFAULT_LOCALE } from '@/const/common/locales';

export const localizePath = (
    to: To,
    locale: string,
    defaultLocale: string = DEFAULT_LOCALE,
): To => {
    if (typeof to !== 'string') {
        return to;
    }

    if (locale === defaultLocale) {
        return to;
    }

    if (to === '/') {
        return `/${locale}`;
    }

    if (to.startsWith(`/${locale}`)) {
        return to;
    }

    return `/${locale}${to}`;
};

```

File: src/components/common/link/Link.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { Link as RouterLink, LinkProps } from 'react-router-dom';
```
Comment (snowflaki) [inline src/components/common/link/Link.tsx]:
```suggestion
import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
```

File: src/components/common/link/Link.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import { useLocale } from '@/hooks/common/use-locale/useLocale';
+import { DEFAULT_LOCALE } from '@/const/common/locales';
+
+type AppLinkProps = Omit<LinkProps, 'to'> & {
+    to: string;
+};
```
Comment (snowflaki) [inline src/components/common/link/Link.tsx]:
```suggestion
type LinkProps = Omit<RouterLinkProps, 'to'> & {
    to: RouterLinkProps['to'];
};
```

File: src/components/common/link/Link.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        if (path.startsWith(`/${currentLanguage}`)) return path;
+
+        if (path === '/') return `/${currentLanguage}`;
+
+        return `/${currentLanguage}${path}`;
+    };
+    return (
+        <RouterLink {...props} to={getLocalizedPath(to)}>
+            {children}
+        </RouterLink>
```
Comment (snowflaki) [inline src/components/common/link/Link.tsx]:
```suggestion
        <RouterLink
            {...props}
            to={localizePath(to, currentLanguage)}
        />
```

File: src/components/common/link/Link.tsx (hunk 6 above)
Comment (weazy12) [reply src/components/common/link/Link.tsx]: add

File: src/components/common/link/Link.tsx (hunk 7 above)
Comment (weazy12) [reply src/components/common/link/Link.tsx]: add

File: src/components/common/link/Link.tsx (hunk 5 above)
Comment (weazy12) [reply src/components/common/link/Link.tsx]: add

File: src/components/common/link/Link.tsx (hunk 4 above)
Comment (weazy12) [reply src/components/common/link/Link.tsx]: add

**Review by snowflaki — APPROVED** [review]

**Review by AndreyBondarchuk — APPROVED** [review]

**Review by pasterik — APPROVED** [review]
